commons/all/src/main/java/org/infinispan/commons/util/IntSets.java

```diff
@@ -178,7 +178,8 @@ public static IntSet mutableSet(int value1, int value2) {
     * @return concurrent set
     */
    public static IntSet concurrentSet(int maxExclusive) {
-      return new ConcurrentSmallIntSet(maxExclusive);
+      // if maxExclusive = 0; then we have an empty set
+      return maxExclusive < 1 ? immutableEmptySet() : new ConcurrentSmallIntSet(maxExclusive);
```

Comment:
I would personally say only return immutable empty set when  maxExclusive == 0. That way we will get the new exception you added above if it is negative.